### PR TITLE
feat(components): [el-rate] rate add size prop

### PR DESCRIPTION
**Pull request by kooriookami**
Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

File: packages/components/rate/src/rate.vue [hunk 1]
```diff
@@ -88,10 +89,17 @@ export default defineComponent({
   setup(props, { emit }) {
     const elForm = inject(elFormKey, {} as ElFormContext)
 
+    const rateSize = useSize()
```
Comment (HerringtonDarkholme) [inline packages/components/rate/src/rate.vue:92]: `rateSize` will return `"default"` so rateSize.value will always be truthy?

File: packages/components/rate/__tests__/rate.spec.ts [hunk 2]
```diff
@@ -12,6 +12,15 @@ describe('Rate.vue', () => {
     expect(stars.length).toEqual(10)
   })
 
+  test('size', () => {
```
Comment (HerringtonDarkholme) [inline packages/components/rate/__tests__/rate.spec.ts:15]: Can we add a test case for `el-rate` without a size prop?

**Review by HerringtonDarkholme — CHANGES_REQUESTED** [review]
Added some comments.

File: packages/components/rate/src/rate.vue (hunk 1 above)
Comment (kooriookami) [reply packages/components/rate/src/rate.vue:92]: Yes, the other components have the same behavior. The buttons also have class `el-button--default`.

**Review by sxzz — APPROVED** [review]

**Review by HerringtonDarkholme — APPROVED** [review]

File: packages/components/rate/src/rate.vue [hunk 3]
```diff
@@ -88,10 +89,17 @@ export default defineComponent({
...
     const currentValue = ref(props.modelValue)
     const hoverIndex = ref(-1)
     const pointerAtLeftHalf = ref(true)
 
+    const prefix = 'el-rate'
+    const rateKls = computed(() => [
+      prefix,
+      rateSize.value ? `${prefix}--${rateSize.value}` : '',
```
Comment (HerringtonDarkholme) [inline packages/components/rate/src/rate.vue:100]: So this `rateSize.value` will be always truthy, right? I wonder if we still need this check.

Comment (JeremyWuuuuu) [reply packages/components/rate/src/rate.vue:100]: Looks like so, I think we can pass the checking part on sizes, it always fallbacks to `default`.

File: docs/en-US/component/rate.md [hunk 4]
```diff
@@ -80,25 +88,26 @@ rate/readonly
...
+| model-value / v-model | binding value                                                                                                                                                                                                                  | number           | —                      | 0                                                                  |
+| max                   | max rating score                                                                                                                                                                                                               | number           | —                      | 5                                                                  |
+| size                  | size of Rate                                                                                                                                                                                                                   | string           | large / default /small | default                                                            |
```
Comment (JeremyWuuuuu) [inline docs/en-US/component/rate.md:95]: Same issue like the other 2 PRs, please add a space after `default /`.

**Review by jw-foss — COMMENTED** [review]
I have left 2 suggestions for you:
1. Update the document
2. Removing the checking for size.

**Review by jw-foss — APPROVED** [review]
LGTM, thanks. /cc @HerringtonDarkholme

**Pull request merged**